packages/labs/gen-wrapper-vue/src/lib/wrapper-module-template-sfc.ts

```diff
@@ -32,67 +36,131 @@ export const wrapperModuleTemplateSFC = (
   ]);
 };
 
-const getEventType = (event: ModelEvent) => event.type?.text || `unknown`;
+//const getEventType = (event: ModelEvent) => event.type?.text || `unknown`;
 
-// TODO(kschaaf): Note, for now we consider all properties optional
-const wrapDefineProps = (props: Map<string, ModelProperty>) =>
-  Array.from(props.values())
-    .map((prop) => `${prop.name}?: ${prop.type?.text}`)
-    .join(',\n');
+const getEventPayloadType = (type: ModelEvent['type']) => {
+  const {text} = type ?? {text: 'unknown'};
+  const {payload} = text.match(/.*<(?<payload>.*)>/)?.groups ?? {
+    payload: text,
+  };
+  return payload;
+};
+
+const defaultEventType = `CustomEvent<unknown>`;
+const isCustomEvent = (type: string) => /^CustomEvent/.test(type);
```

Comment:
We should go with the same approach in React & Vue (either pass detail for CustomEvent, or just pass the Event in all cases). Per convo, maybe just start with always passing the event.

Reply:
Done.